llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -2498,6 +2499,15 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
         MSV->setOrigin(I, Origin);
       }
     }
+
+    /// Store the current combined value at the specified origin
+    /// location.
+    void DoneAndStoreOrigin(TypeSize TS, Value *OriginPtr) {
+      if (MSV->MS.TrackOrigins) {
+        assert(Origin);
+        MSV->paintOrigin(IRB, Origin, OriginPtr, TS, Align(1));
```

Comment:
Origin aligment must be 4+

`const Align OriginAlignment = std::max(kMinOriginAlignment, Alignment);`
and I suspect this instruction guaranty some `Alignment`?

Reply:
I guess LGTM, if we don't know Ptr Alignment

Reply:
Updated to kMinOriginAlignment in https://github.com/llvm/llvm-project/pull/99360/commits/2d0abae869d0c565dc6230ec8adbaf5067edfc03

Reply:
Changed back to 1-byte alignment in https://github.com/llvm/llvm-project/pull/99360/commits/af86d86e1d00f09c4e8c8f425a5bca6522636820 per offline discussion